create-update.sh

```diff
@@ -89,58 +73,35 @@ fi
 export BUNDLEREPO="$BUNDLE_DIR"
 
 if [ ! -d "$STATE_DIR/www/version/format$FORMAT" ]; then
-    sudo -E mkdir -p "$STATE_DIR/www/version/format$FORMAT/"
+	sudo -E mkdir -p "$STATE_DIR/www/version/format$FORMAT/"
 fi
 
 # step 1: create update content for current mix
 sudo -E "$PREFIX"swupd_create_update -S "$STATE_DIR" --minversion "$MINVERSION" -F "$FORMAT" --osversion $MIXVER
 
 # we only need the full chroot from this point on, so cleanup the others
 if [  "$KEEP_CHROOTS" -eq 0 ]; then
-		for BUNDLE in $(ls "$BUNDLEREPO" | grep -v "^full"); do
-			sudo rm -rf "$STATE_DIR/image/$MIXVER/$BUNDLE"
-		done
+	clean_chroots
 fi
 
 # step 1.5: sign the Manifest.MoM that was just created
-if [ -f "ClearLinuxRoot.pem" ]; then
-	sudo openssl smime -sign -binary -in update/www/$MIXVER/Manifest.MoM \
-		-signer ClearLinuxRoot.pem -inkey private.pem \
-		-outform DER -out update/www/$MIXVER/Manifest.MoM.sig
-fi
+sign_manifest_mom
 
 # step 2: create fullfiles
 sudo -E "$PREFIX"swupd_make_fullfiles -S "$STATE_DIR" $MIXVER
 
 # step 3: create zero packs
-MOM="$STATE_DIR/www/$MIXVER/Manifest.MoM"
-if [ ! -e ${MOM} ]; then
-    error "no ${MOM}"
-fi
-BUNDLE_LIST=$(cat ${MOM} | awk -v V=${MIXVER} '$1 ~ /^M\./ && $3 == V { print $4 }')
-# NOTE: for signing, pass the --signcontent option to swupd_make_pack.
-# Signing is currently disabled until there are new test certs ready.
 if [ $ZEROPACKS -eq 1 ]; then
-	for BUNDLE in $BUNDLE_LIST; do
-		sudo -E "$PREFIX"swupd_make_pack -S "$STATE_DIR" 0 $MIXVER $BUNDLE &
-	done
-
-	for job in $(jobs -p); do
-	    wait ${job}
-	    RET=$?
-	    if [ "$RET" != "0" ]; then
-		error "zero pack subprocessor failed"
-	    fi
-	done
+	sudo -E mixer-pack-maker.sh --to $MIXVER -S "$STATE_DIR"
 fi
 
 # step 4: hardlink relevant dirs
 sudo -E "hardlink" -f "$STATE_DIR/image/$MIXVER"/
 
 # step 5: update latest version
 if [ $NOPUBLISH -eq 0 ]; then
-	sudo cp "$PWD/.mix-version" "$STATE_DIR/image/LAST_VER"
-	sudo cp "$PWD/.mix-version" "$STATE_DIR/www/version/format$FORMAT/latest"
+	sudo -E echo "$MIXVER" > "$STATE_DIR/image/LAST_VER"
+	sudo -E echo "$MIXVER" > "$STATE_DIR/www/version/format$FORMAT/latest"
```

Comment:
`echo` might be a shell builtin, in which case I would expect these two commands to fail. I would suggest using `sh -c ...` instead.